src/library/scala/collection/GenTraversableOnce.scala

```diff
@@ -67,6 +67,20 @@ trait GenTraversableOnce[+A] extends Any {
     */
   def foreach[U](f: A => U): Unit
 
+  /** Tests whether this $coll is known to have a finite size.
+    *  All strict collections are known to have finite size. For a non-strict
+    *  collection such as `Stream`, the predicate returns `'''true'''` if all
+    *  elements have been computed. It returns `'''false'''` if the stream is
+    *  not yet evaluated to the end. Non-empty Iterators usually return
+    *  `'''false'''` even if they were created from a collection with a known
+    *  finite size.
+    *
+    *  Note: many collection methods will not work on collections of infinite sizes
+    *  (but they do not require this method to return `'''true'''`).
```

Comment:
Do you mean something like (but will work on finite collections whose size is not known)?  Thinking about what it would mean to require this method to return true seems like it requires more mental gymnastics than strictly necessary.  (Assuming that the person thinking about it has never seen this method before.)